choral/libs/choral-reactive/src/main/java/choral/reactive/ReactiveServer.java

```diff
@@ -23,27 +23,46 @@ public class ReactiveServer
 
     private final HashSet<Integer> knownSessionIDs = new HashSet<>();
 
+    // INVARIANTS:
+    // 1. sendQueue contains session if and only if recvQueue contains session.
+    // 2. if sendQueue is non-empty, then recvQueue is empty; and vice versa.
+    // 3. if a sessionID is in knownSessionIDs, then the session is in telemetrySessionMap.
+
+    /** Maps a sessionID and a sender name to a queue of messages. The next receive operation will consume the first message in the queue. */
     private final HashMap<Session, LinkedList<Serializable>> sendQueue = new HashMap<>();
+    /** Maps a sessionID and a sender name to a queue of futures. The next message will be fed to the first future in the list. */
     private final HashMap<Session, LinkedList<CompletableFuture<Serializable>>> recvQueue = new HashMap<>();
+    /** Maps a sessionID to a TelemetrySession. */
     private final HashMap<Integer, TelemetrySession> telemetrySessionMap = new HashMap<>();
 
     private final String serviceName;
     private final NewSessionEvent newSessionEvent;
     private final OpenTelemetrySdk telemetry;
     private final ServerConnectionManager connectionManager;
 
+    /**
+     * Creates a ReactiveServer, using {@link ServerConnectionManager} for the connection.
+     * Invoke {@link ServerConnectionManager#listen(String)} to start listening.
```

Comment:
The proper way to listen is to call `ReactiveServer#listen(String)` rather than `ServerConnectionManager#listen(String)` directly